lib/Common/Memory/CustomHeap.cpp

```diff
@@ -251,6 +251,10 @@ Allocation* Heap::Alloc(size_t bytes, ushort pdataCount, ushort xdataSize, bool
         {
             page = AllocNewPage(bucket, canAllocInPreReservedHeapPageSegment, isAnyJittedCode, isAllJITCodeInPreReservedRegion);
         }
+        else if (!canAllocInPreReservedHeapPageSegment && isAnyJittedCode)
+        {
+            *isAllJITCodeInPreReservedRegion = false;
+        }
```

Comment:
Well, it's a real issue that my change exposed, probably by causing more re-jitting.